egs/wsj0-mix-var/Multi-Decoder-DPRNN/metrics.py

```diff
@@ -0,0 +1,90 @@
+import torch
+import torch.nn as nn
+from asteroid.losses import PITLossWrapper, pairwise_neg_sisdr
+from torch.nn.modules.loss import _Loss
+from scipy.optimize import linear_sum_assignment
+
+
+class PairwiseNegSDR_Loose(_Loss):
+    """
+    Same as asteroid.losses.PairwiseNegSDR, but supports speaker number mismatch
+    """
+
+    def __init__(self, sdr_type, zero_mean=True, take_log=True, EPS=1e-8):
+        super(PairwiseNegSDR_Loose, self).__init__()
+        assert sdr_type in ["snr", "sisdr", "sdsdr"]
+        self.sdr_type = sdr_type
+        self.zero_mean = zero_mean
+        self.take_log = take_log
+        self.EPS = EPS
+
+    def forward(self, est_targets, targets):
```

Comment:
Is there a way not to duplicate all this code, but subclass it, with minor modif.

Reply:
The main problem is the line assert targets.size() == est_targets.size() in your original code, as it does not support variable number of speakers.